### PR TITLE
Reintroduce readtable(), writetable() and string-literal macros, keeping them deprecated

### PR DESCRIPTION
**Pull request by nalimilan**
This will ease the transition to CSV, without introducing a dependency on it
and without breakage due to behavior changes.

I tested this locally, but it's not covered by the test suite.

Fixes https://github.com/JuliaData/DataFrames.jl/issues/1240.

### CODE REVIEW

File: src/deprecated.jl [hunk 1]
```diff
@@ -17,3 +17,1252 @@ import Base: keys, values, insert!
...
+
+* `::DataFrame`
+
+### Examples
+
+```julia
+df = DataFrame(A = 1:10)
+writetable("output.csv", df)
+writetable("output.dat", df, separator = ',', header = false)
+writetable("output.dat", df, quotemark = '\', separator = ',')
```
Comment (cjprybol) [inline src/deprecated.jl]: I think quotemark here needs to be escaped, `quotemark = '\'` -> `quotemark = '\\'`, although I couldn't find any errors in the CI to confirm that

**Comment by cjprybol** [comment]
This is just copied and pasted from earlier source code, right? If so, then I don't think we should worry about adding the tests back as well.

I don't love the idea of pointing people towards CSV while https://github.com/JuliaData/CSV.jl/issues/109 is unresolved.

**Comment by nalimilan** [comment]
> This is just copied and pasted from earlier source code, right? If so, then I don't think we should worry about adding the tests back as well.

Yeah, that's copy/pasted, except for the place where `DataArray` columns where created (but changes were very limited). 

> I don't love the idea of pointing people towards CSV while JuliaData/CSV.jl#109 is unresolved.

We're not going to tag a release immediately anyway, so that leaves time to fix that nasty bug.

File: src/deprecated.jl (hunk 1 above)
Comment (nalimilan) [reply src/deprecated.jl]: Indeed, it must have been there for a very long time. Fixed (I think the correct value is `'\''`).

**Pull request merged**